### ISSUE TITLE
[web-animations-1] Calling `cancel()` should dispatch a cancel event when the `startTime` is cleared

### ISSUE THREAD

**Issue by birtles** [w3c/csswg-drafts#2106]
In the procedure to [cancel an animation](https://drafts.csswg.org/web-animations-1/#cancel-an-animation), we first check if the state is idle, and don't dispatch cancel events or reject the finished promise if it is.

However, it's possible to be in a state with a valid `startTime` but still be idle (e.g. when the animation has no timeline, or has an inactive timeline, and therefore the current time is unresolved). In that case, calling `cancel()` will clear the `startTime` and I would expect that it should also dispatch a cancel event to indicate that something was canceled (notably the `startTime`).

In other words, the only time calling `cancel()` should _not_ dispatch a cancel event is when it is truly a no-op. But perhaps it should dispatch it then too?

An example of the above case:
```js
const animation = new Animation(null, null);
animation.startTime = 100;
// At this point animation.playState === 'idle' because it has no timeline
animation.cancel();
// The above will reset animation.startTime to null
```

**Comment by birtles** [comment]
Thinking about this further, I think it probably does make sense to only dispatch cancel events when we actually change state. The reason is that the cancel event is expected to be used when the animation ceases to have an effect on its target, which corresponds to when its `currentTime` becomes null. So I'm going to close this for now.
